src/common/conditional_compilation/include/openvino/cc/pass/itt.hpp

```diff
@@ -35,9 +42,45 @@ OV_CC_DOMAINS(ov_pass);
 
 #    define RUN_ON_MODEL_SCOPE(region) MATCHER_SCOPE_(ov_pass, OV_PP_CAT(region, _run_on_model))
 
+#    define ADD_MATCHER_NO_OBJ_1(region, ...) add_matcher<region>(__VA_ARGS__);
+#    define ADD_MATCHER_NO_OBJ_0(region, ...)
+#    define ADD_MATCHER_FOR_THIS(nspace, region, ...)                                          \
+        OV_PP_CAT(ADD_MATCHER_NO_OBJ_, OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(ov_pass, _, region))) \
+        (nspace::region, __VA_ARGS__)
+#    define ADD_MATCHER_FOR_THIS_WITHOUT_NSPACE(region, ...)                                   \
+        OV_PP_CAT(ADD_MATCHER_NO_OBJ_, OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(ov_pass, _, region))) \
+        (region, __VA_ARGS__)
+#    define ADD_MATCHER_OBJ_1(obj, region, ...) obj->add_matcher<region>(__VA_ARGS__);
+#    define ADD_MATCHER_OBJ_0(obj, region, ...)
+#    define ADD_MATCHER(obj, nspace, region, ...)                                           \
+        OV_PP_CAT(ADD_MATCHER_OBJ_, OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(ov_pass, _, region))) \
+        (obj, nspace::region, __VA_ARGS__)
+#    define ADD_MATCHER_WITHOUT_NSPACE(obj, region, ...)                                    \
+        OV_PP_CAT(ADD_MATCHER_OBJ_, OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(ov_pass, _, region))) \
+        (obj, region, __VA_ARGS__)
+#    define REGISTER_PASS_1(obj, region, ...) obj.register_pass<region>(__VA_ARGS__);
+#    define REGISTER_PASS_0(obj, region, ...)
+#    define REGISTER_PASS(obj, nspace, region, flag, ...)                                                  \
+        OV_PP_CAT(REGISTER_PASS_, OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(ov_pass, _, OV_PP_CAT(region, flag)))) \
```

Comment:
I think here we can check all additional suffix like `_run_on_function`, `run on model` and etc.

Reply:
My point is how to check the suffix?
Here is my proposal.
1: Revert to my original implementation, for example: Add more 2 MACRO define.
  #    define REGISTER_PASS(obj, region, ...)
  #    define REGISTER_PASS_RUN_ON_FUN(obj, region, ...)
  #    define REGISTER_PASS_RUN_ON_MODEL(obj, region, ...)

What do you think? or could you provide more your idea detail about how to check the suffix?

Reply:
@ilyachur For this problem, I still need your advice.

Reply:
Why cannot you have 1 high level macro:
```
REGISTER_PASS
```
And inside we defines 3 marcos: PASS, PASS_run_on_model, PASS_run_on_function

Reply:
Got, your advice is similar with my second proposal.

Reply:
But I don't want to have run_on_model or something like that on user level. User shouldn't know about it

Reply:
@ilyachur Got, customer only need to touch "REGISTER_PASS", because there are only 3 possibilities, I can list all combinations of the 3 possibilities.
For example: 
#define OR_000 0
#define OR_001 1
#define OR_010 1
#define OR_011 1
#define OR_100 1
#define OR_101 1
#define OR_110 1
#define OR_111 1

The any one of the combination is positive, we will disable it.
Please my latest update for detail.
HaHa, hope it doesn't shock you.

Reply:
Take "REGISTER_PASS(manager, ConstantFolding)" as example.
My original idea is that finding "ConstantFolding"'s defination in file "conditional_compilation_gen.h", however we can find 3 situations,  
1: "#define ov_pass_ConstantFolding 1";
2: "#define ov_pass_ConstantFolding_run_on_model 1";
3: "#define ov_pass_ConstantFolding_run_on_function 1";
So now I will judge the all 3 situations for latest implement, any one of the situtations is true, we will disable it.

Reply:
It is not a problem, because in real life we cannot have `ConstantFolding_run_on_model` and `ConstantFolding_run_on_function` at the same time